dspace-server-webapp/src/test/java/org/dspace/app/rest/signposting/controller/LinksetRestControllerIT.java

```diff
@@ -0,0 +1,677 @@
+                .withTitle("Workspace Item")
+                .build();
+        itemService.addMetadata(context, workspaceItem.getItem(), "dc", "identifier", "doi", Item.ANY, doi);
+        context.restoreAuthSystemState();
+
+        getClient().perform(get("/signposting/linksets/" + workspaceItem.getItem().getID() + "/json"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.linkset",
+                        Matchers.hasSize(1)))
+                .andExpect(jsonPath("$.linkset[0].cite-as[0].href",
+                        Matchers.hasToString(MessageFormat.format(doiPattern, doi))));
+    }
+
+    @Test
+    public void findOneWithdrawnItemJsonLinksets() throws Exception {
+        context.turnOffAuthorisationSystem();
+        Item item = ItemBuilder.createItem(context, collection)
+                .withTitle("Withdrawn Item")
+                .withMetadata("dc", "identifier", "doi", doi)
+                .withdrawn()
+                .build();
+        context.restoreAuthSystemState();
+
+        getClient().perform(get("/signposting/linksets/" + item.getID() + "/json"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.linkset",
+                        Matchers.hasSize(1)))
+                .andExpect(jsonPath("$.linkset[0].cite-as[0].href",
+                        Matchers.hasToString(MessageFormat.format(doiPattern, doi))));
+    }
+
+    @Test
+    public void findOneEmbargoItemJsonLinksets() throws Exception {
+        context.turnOffAuthorisationSystem();
+        Item item = ItemBuilder.createItem(context, collection)
+                .withTitle("Withdrawn Item")
+                .withMetadata("dc", "identifier", "doi", doi)
+                .withIssueDate("2017-11-18")
+                .withEmbargoPeriod("2 week")
+                .build();
+        context.restoreAuthSystemState();
+
+        getClient().perform(get("/signposting/linksets/" + item.getID() + "/json"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.linkset",
+                        Matchers.hasSize(1)))
+                .andExpect(jsonPath("$.linkset[0].cite-as[0].href",
+                        Matchers.hasToString(MessageFormat.format(doiPattern, doi))));
+    }
+
+    @Test
+    public void findOneRestrictedItemJsonLinksets() throws Exception {
+        context.turnOffAuthorisationSystem();
+        Group internalGroup = GroupBuilder.createGroup(context)
+                .withName("Internal Group")
+                .build();
+        Item item = ItemBuilder.createItem(context, collection)
+                .withTitle("Withdrawn Item")
+                .withMetadata("dc", "identifier", "doi", doi)
+                .withReaderGroup(internalGroup)
+                .build();
+        context.restoreAuthSystemState();
+
+        getClient().perform(get("/signposting/linksets/" + item.getID() + "/json"))
+                .andExpect(status().isOk())
```

Comment:
Similar to above. Should not be possible to get a restricted Item in the signposting links.  This should return a 401.  If you are logged in & have access to the item it could return 200.  But, it's also arguable that signposting should only work for **public content**.

Reply:
Fixed